oauth2_http/java/com/google/auth/oauth2/ServiceAccountJwtAccessCredentials.java

```diff
@@ -255,15 +250,44 @@ public long read() {
             }
         )
         .build(
-            new CacheLoader<URI, String>() {
+            new CacheLoader<JwtCredentials.Claims, JwtCredentials>() {
               @Override
-              public String load(URI key) throws Exception {
-                return generateJwtAccess(key);
+              public JwtCredentials load(JwtCredentials.Claims claims) throws Exception {
+                return JwtCredentials.newBuilder()
+                    .setPrivateKey(privateKey)
+                    .setPrivateKeyId(privateKeyId)
+                    .setClaims(claims)
+                    .setLifeSpanSeconds(LIFE_SPAN_SECS)
+                    .setClock(clock)
+                    .build();
               }
             }
         );
   }
 
+  /**
+   * Returns a new JwtCredentials instance with modified claims.
+   *
+   * @param newClaims New claims. Any unspecified claim fields will default to the the current
+   *        values.
+   * @return new credentials
+   */
+  public JwtCredentials withClaims(JwtCredentials.Claims newClaims) {
```

Comment:
Should this be named `jwtWithClaims`? If we add this method to `ServiceAccountCredentials` and `ImpersonatedCredentials` (both which also can generate JWT tokens), then maybe it's not clear what we're building.

Reply:
Additionally, we may be implementing something similar with IdToken so we'd create a method like `idTokenWithAudience(String audience)`